### ISSUE TITLE
Still immutable error when using --force

### ISSUE THREAD

**Issue by b4nst** [stefanprodan/timoni#209]
As I understood #155 bumped `github.com/fluxcd/pkg/ssa` version which should have resolved resources not recreated when using `--force` arg. However I'm still facing the issue with some resources:

```sh
11:20AM ERR IAMServiceAccount/banst/opkg-registry dry-run failed: admission webhook "deny-immutable-field-updates.cnrm.cloud.google.com" denied the request: cannot make changes to immutable field(s): [spec.resourceID]; please refer to our troubleshooting doc: https://cloud.google.com/config-connector/docs/troubleshooting
```

Is there something we can do to catch that and recreate the resource when using `--force` or `"action.timoni.sh/force": "enabled"` annotation?

```shell
timoni version
api: timoni.sh/v1alpha1
client: 0.14.0
cue: 0.6.0
```

**Comment by stefanprodan** [comment]
Yeah that error does not come from Kubernetes CEL but some GKE webhook and the error doesn't match the regex we use for CEL. https://github.com/fluxcd/pkg/blob/6dc2f174f41fb859afafc0d6f28d58d6a610e606/ssa/utils.go#L288C26-L288C26

**Comment by b4nst** [comment]
Any chance we could check custom webhooks too or would that be too much work? I'm asking that totally out of context since I did not check the code. Sorry if that sounds dumb

**Comment by stefanprodan** [comment]
Doing string matching on error messages is really bad, especially for something like force apply which has the potential to wipe your cluster clean if it matches something wrong. I could maybe expose this somehow in the SSA package and allow Timoni users to go YOLO with regexes while risking major incidents. How ofter do you run into this, is this an exception or your are changing immutable fields in GPC objects all the time?

**Comment by b4nst** [comment]
Sooo, that's the thing with config connector. A ton of stuff is immutable and needs recreation of the resource (because the Google resource itself needs to be recreated). So to answer your question, yes this is something I run into way more often that I would like 😅. They even have immutable fields that should not be immutable (see https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/670#issuecomment-1179301208 for example).
